internal/common/rules/basicruleevaluator.go

```diff
@@ -167,3 +167,32 @@ func BasicRealValuedRuleEvaluator(ruleName string) (bool, float64, error) {
 	}
 	return false, 0, errors.Errorf("rule name: '%v' provided doesn't exist in global rule list", ruleName)
 }
+
+// BasicLinkedRuleEvaluator evaluates linked rules using the above two evaluators
+func BasicLinkedRuleEvaluator(ruleName string) (bool, error) {
+	if rm, ok := AvailableRules[ruleName]; ok {
+		link := rm.Link
+		if !link.Linked {
+			return BasicBooleanRuleEvaluator(ruleName)
+		} else {
+			if link.LinkType == ParentFailAutoRulePass {
+				childRule := link.LinkedRule
+				parentPass, parentErr := BasicBooleanRuleEvaluator(ruleName)
+				if parentErr != nil {
+					return false, errors.Errorf("Parent Rule errored out wiht : %v", parentErr)
+				}
+				if !parentPass {
+					return true, nil
+				}
+				childPass, childErr := BasicBooleanRuleEvaluator(childRule)
```

Comment:
Here we evaluate the child rule even if it is not in play, is that right or am I misunderstanding?

Reply:
Yes, all the rule evaluators feed off the available rules cache to allow for islands to look at prospective rules. It's up to us to check whether the rules are in play before we call this.